doc/dev/tests-advanced.md

```diff
@@ -76,38 +75,42 @@ class TestTablesUnit(TablesTestMixin):
 
+**Note:** if test methods in a class use management-plane resource preparers like the following, that test class must inherit from [AzureMgmtRecordedTestCase](https://github.com/Azure/azure-sdk-for-python/blob/main/tools/azure-sdk-tools/devtools_testutils/mgmt_recorded_testcase.py) instead of `AzureRecordedTestCase`.
+
 | Preparer | `create_resource` | `remove_resource` |
 |-|-|-|
-| Resource Group | [link](https://github.com/Azure/azure-sdk-for-python/blob/main/tools/azure-sdk-tools/devtools_testutils/resource_testcase.py#L57-L85) | [link](https://github.com/Azure/azure-sdk-for-python/blob/main/tools/azure-sdk-tools/devtools_testutils/resource_testcase.py#L87-L99) |
-| Storage Account | [link](https://github.com/Azure/azure-sdk-for-python/blob/main/tools/azure-sdk-tools/devtools_testutils/storage_testcase.py#L53-L102) | [link](https://github.com/Azure/azure-sdk-for-python/blob/main/tools/azure-sdk-tools/devtools_testutils/storage_testcase.py#L104-L107) |
-| KeyVault | [link](https://github.com/Azure/azure-sdk-for-python/blob/main/tools/azure-sdk-tools/devtools_testutils/keyvault_preparer.py#L84-L131) | [link](https://github.com/Azure/azure-sdk-for-python/blob/main/tools/azure-sdk-tools/devtools_testutils/keyvault_preparer.py#L133-L138) |
+| Resource Group | [link](https://github.com/Azure/azure-sdk-for-python/blob/a96aee38786daddf4fc6fa0e487f243ed8c9fde9/tools/azure-sdk-tools/devtools_testutils/resource_testcase.py#L71) | [link](https://github.com/Azure/azure-sdk-for-python/blob/a96aee38786daddf4fc6fa0e487f243ed8c9fde9/tools/azure-sdk-tools/devtools_testutils/resource_testcase.py#L109) |
+| Storage Account | [link](https://github.com/Azure/azure-sdk-for-python/blob/20e5bdca511fed4358a8d33a0beeb48cb6c3d8a1/tools/azure-sdk-tools/devtools_testutils/storage_testcase.py#L73) | [link](https://github.com/Azure/azure-sdk-for-python/blob/20e5bdca511fed4358a8d33a0beeb48cb6c3d8a1/tools/azure-sdk-tools/devtools_testutils/storage_testcase.py#L112) |
+| KeyVault | [link](https://github.com/Azure/azure-sdk-for-python/blob/c4cbcee52fbe486472c6b28af68f751dd3e2d016/tools/azure-sdk-tools/devtools_testutils/keyvault_preparer.py#L88) | [link](https://github.com/Azure/azure-sdk-for-python/blob/c4cbcee52fbe486472c6b28af68f751dd3e2d016/tools/azure-sdk-tools/devtools_testutils/keyvault_preparer.py#L134) |
 
 
 ## Examples with Preparers
 
-### Example 2: Basic Preparer Usage with Storage
+### Example 1: Basic Preparer Usage with Storage
 
 ```python
 import os
 import pytest
 
 from azure.data.tables import TableServiceClient
 from devtools_testutils import (
-    AzureTestCase,
+    AzureMgmtRecordedTestCase,
+    recorded_by_proxy,
     ResourceGroupPreparer,
-    StorageAccountPreparer
+    StorageAccountPreparer,
 )
 
-class ExampleStorageTestCase(AzureTestCase):
+class TestExampleStorage(AzureMgmtRecordedTestCase):
```

Comment:
Did we mean this to be AzureMgmtRecordedTestCase or AzureRecordedTestCase?

Just clarifying for my own edification.

Reply:
This is meant to be AzureMgmtRecordedTestCase because of the use of ResourceGroupPreparer (which needs a management client available from the test class). It'll technically be kind of outdated soon since Storage is moving to `test-resources.json` as part of the test proxy migration, but I was thinking that this helps to emphasize the point about when AzureMgmtRecordedTestCase would be necessary